odoo/tools/pdf.py

```diff
@@ -19,3 +50,258 @@ def merge_pdf(pdf_data):
     merged_pdf = _buffer.getvalue()
     _buffer.close()
     return merged_pdf
+
+
+class OdooPdfFileWriter(PdfFileWriter):
+
+    def __init__(self, *args, **kwargs):
+        """
+        Override of the init to initialise additional variables.
+        :param pdf_content: if given, will initialise the reader with the pdf content.
+        """
+        super().__init__(*args, **kwargs)
+        self._is_pdfa = False
+        self._attachments_data = []
+        self._metadata = None
+
+    def cloneReaderDocumentRoot(self, reader):
+        """
+        Override from the base class to set the header properly, as well as give an id to the pdf.
+        The base class would always write PDF-1.3 without the additional bytes required by PDF/A, and always
+        remove the id.
+        """
+        super().cloneReaderDocumentRoot(reader)
+        # Set the PDF version to 1.7 (as PDF/A-3 are in version 1.7) and append some more bytes required by PDF/A
+        self._header = "%PDF-1.7\n%����".encode()
```

Comment:
Why are you not using a bytes literal and binary escapes? The last 4 characters are completely garbled.

The behaviour also seems debatable, what will happen to existing clients who don't care, is 1.7 and its "some more bytes" fully backwards compatible?